### PR TITLE
fix: yamllint errors, typo in kubebuilder presubmit; dind docs

### PR DESCRIPTION
**Pull request by alexeldeib**
Docker in docker requires that the jobs start up using the provided runner in `/usr/local/bin/runner.sh`. Update the doc comment to reflect the correct filepath, and update Kubebuilder to use the runner.

### CODE REVIEW

**Comment by alexeldeib** [comment]
sorry to spam you guys

/assign @mengqiy @DirectXMan12

**Comment by alexeldeib** [comment]
buuuunch of yamllint errors in master which looked like a bad copy paste/automatic application of privileged security context where necessary + resource limits? Fixed 36 yamllint errors in unrelated code.

**Comment by alexeldeib** [comment]
guess i'll follow the bot's directions now that it's not a one-liner 

/assign @spiffxp

File: config/prow/config.yaml [hunk 1]
```diff
@@ -590,7 +590,7 @@ presets:
 # docker-in-docker (with images/bootstrap) preset
 # NOTE: using this also requires using that image,
 # ensuring you run your test under either the ENTRYPOINT or:
-# /usr/local/bin/runner my-test-command --foo --bar
+# /usr/local/bin/runner.sh my-test-command --foo --bar
```
Comment (alexeldeib) [inline config/prow/config.yaml]: n.b.

**Comment by justaugustus** [comment]
Thanks for the fixes, @alexeldeib!
/lgtm
/approve
/assign @cblecker @krzyzacy

**Comment by cblecker** [comment]
/lgtm
/approve

Thanks for this! 🚀

**Pull request merged**